apps/08jquery-towers-of-hanoi/script.js

```diff
@@ -1,5 +1,49 @@
 'use strict';
+var block=null;
 
 $(document).ready(function() {
   // Put app logic here
+  $('[data-stack]').click(dataStackWasClicked);
+  
 });
+
+function dataStackWasClicked() {
+  if (block===null) {
+    //Get the last child element of the data stack element.
+    var lastChild = $(this).children().last();
+    if (lastChild.length!=0)
+      {
+      //If it's not empty. Detach it and store it in the block variable.
+      block = lastChild.detach();
+      }
+    }
+  else
+    {
+    //There is an element (a data-block pulled from the data-stack div) contained in the block variable.
+    var lastBlockOfStack=$(this).children().last();
```

Comment:
Don't create two different variables for the same logic inside one function (See line 13)